src/makielayout/blocks/textbox.jl

```diff
@@ -210,6 +213,24 @@ function initialize_block!(tbox::Textbox)
 
     on(events(scene).keyboardbutton, priority = 60) do event
         if tbox.focused[]
+            ctrl_v = (Keyboard.left_control | Keyboard.right_control) & Keyboard.v
+            if ispressed(scene, ctrl_v)
+                local content::String = ""
+                try
+                    content = clipboard()
+                catch err
+                    @warn "Pasting failed: $err"
+                    return Consume(false)
+                end
+
+                if all(char -> is_allowed(char, tbox.restriction[]), content)
+                    foreach(char -> insertchar!(char, cursorindex[] + 1), content)
+                    return Consume(true)
+                else
+                    return Consume(false)
```

Comment:
Shouldn't there be some sort of informative warning if pasting fails because some chars fall under the restriction? Such as it is, nothing would happen, right?

Reply:
I thought about this... My conclusion was, that this is a user restriction, so the user should chose, right?
And if the user wants to warn other users, they could include a warning in their `restriction` function...

Reply:
it's mostly for stuff like "only allow numbers" but of course the validation function should notify about that anyway. maybe it doesn't matter

Reply:
haven't used restriction in practice yet